generate_and_apply_patch.sh

```diff
@@ -4,13 +4,25 @@ set -o pipefail
 
 BOOT=${1:-/boot}
 IRFS_HOOK="/etc/initramfs-tools/hooks/acpi_override.sh"
+OSID=$(grep ID= /etc/os-release | cut -d '=' -f2)
 cd `mktemp -d`
 
-# Make sure we have required tools on systems with apt
-if [ -x $(which apt-get) ]; then
-	echo "[*] Installing required tools"
-	sudo apt-get -y install acpica-tools cpio
-fi
+# Make sure we have required tools on systems
+case $OSID in
+	debian)
+		if [ -x $(which apt-get) ]; then
+			echo "[*] Installing required tools"
```

Comment:
could you add an echo to the `arch` section or make a global "trying to install dependencies".

Reply:
I believe this approach is suboptimal because we do not care about the OS version, we care about the package manager being used so we can install the proper tools. There are several distros which are technically different and report different names but rely on the same package manager, ex. Ubuntu, Elementary OS, and Debian all use `apt-get` or Antergos and Arch both use `pacman.` By explicitly checking for `Debian` vs `apt-get`, we unintentionally leave out other apt-get based distros that the script would technically work fine with because the OSID doesn't match.

Reply:
In that case the distro in question should be tested and added to the script accordingly. apt-get is not sufficient due to the packages potentially being named diffenrently. In addition to that, further down the debian branch does some fiddling with the initramfs which is ommitted for Arch, for example. I don't know if all apt-get-using distros use the same tooling as debian?

But that's merely academic, as there is a much better fix out there from Lenovo by now than this script.